helix-term/src/main.rs

```diff
@@ -115,25 +115,69 @@ FLAGS:
 
     setup_logging(logpath, args.verbosity).context("failed to initialize logging")?;
 
-    let config_dir = helix_loader::config_dir();
-    if !config_dir.exists() {
-        std::fs::create_dir_all(&config_dir).ok();
-    }
-
     helix_loader::initialize_config_file(args.config_file.clone());
 
-    let config = match std::fs::read_to_string(helix_loader::config_file()) {
-        Ok(config) => toml::from_str(&config)
-            .map(helix_term::keymap::merge_keys)
-            .unwrap_or_else(|err| {
-                eprintln!("Bad config: {}", err);
-                eprintln!("Press <ENTER> to continue with default config");
-                use std::io::Read;
-                let _ = std::io::stdin().read(&mut []);
-                Config::default()
-            }),
-        Err(err) if err.kind() == std::io::ErrorKind::NotFound => Config::default(),
-        Err(err) => return Err(Error::new(err)),
+    // We need the config, to get the editor.security.[load|confirm]_local_config flags
+    let root_config = std::fs::read_to_string(helix_loader::config_file())
+        .map(|config| toml::from_str(&config))?
+        .unwrap_or_default();
+
+    // Now, let's load config based on a PathBuf
+    // We will keep it toml::Value so we can use helix_loader::merge_toml_values
+    // This will respect editor.security settings
+    let load_config = |config_path: std::path::PathBuf| {
+        if std::path::Path::new(&config_path).exists() {
```

Comment:
```suggestion
        if config_path.exists() {
```
This auto deref to `Path`, so don't need to manually create new `Path`.